tests/components/bryant_evolution/test_climate.py

```diff
@@ -0,0 +1,348 @@
+    mock_client.set_cooling_setpoint.assert_called_once_with(80)
+    mock_client.set_heating_setpoint.assert_called_once_with(70)
+
+
+async def test_set_fan_mode(
+    hass: HomeAssistant, mock_evolution_entry: MockConfigEntry
+) -> None:
+    """Test that setting fan mode works."""
+    mock_client = mock_evolution_entry.runtime_data[(1, 1)]
+    fan_modes = ["auto", "low", "med", "high"]
+    for mode in fan_modes:
+        # Change the fan mode, pausing reads to the device so that we
+        # verify that changes are locally committed.
+        data = {ATTR_FAN_MODE: mode}
+        data[ATTR_ENTITY_ID] = "climate.system_1_zone_1"
+        with disable_auto_entity_update():
+            await hass.services.async_call(
+                "climate", SERVICE_SET_FAN_MODE, data, blocking=True
+            )
+            await hass.async_block_till_done()
+        assert (
+            hass.states.get("climate.system_1_zone_1").attributes[ATTR_FAN_MODE] == mode
+        )
+        mock_client.set_fan_mode.assert_called_with(mode)
+
+
+async def test_set_hvac_mode(
+    hass: HomeAssistant, mock_evolution_entry: MockConfigEntry
+) -> None:
+    """Test that setting HVAC mode works."""
+    mock_client = mock_evolution_entry.runtime_data[(1, 1)]
+    hvac_modes = ["heat_cool", "heat", "cool", "off"]
+    for mode in hvac_modes:
+        # Change the mode, pausing reads to the device so that we
+        # verify that changes are locally committed.
+        data = {ATTR_HVAC_MODE: mode}
+        data[ATTR_ENTITY_ID] = "climate.system_1_zone_1"
+        with disable_auto_entity_update():
+            await hass.services.async_call(
+                "climate", SERVICE_SET_HVAC_MODE, data, blocking=True
+            )
+            await hass.async_block_till_done()
+        evolution_mode = "auto" if mode == "heat_cool" else mode
+        assert hass.states.get("climate.system_1_zone_1").state == evolution_mode
+        mock_client.set_hvac_mode.assert_called_with(evolution_mode)
+
+
+@pytest.mark.parametrize("curr_temp", [62, 70, 80])
+async def test_read_hvac_action_heat_cool(
+    hass: HomeAssistant,
+    mock_evolution_entry: MockConfigEntry,
+    freezer: FrozenDateTimeFactory,
+    curr_temp: int,
+) -> None:
+    """Test that we can read the current HVAC action in heat_cool mode."""
+    htsp = 68
+    clsp = 72
+
+    mock_client = mock_evolution_entry.runtime_data[(1, 1)]
+    mock_client.read_heating_setpoint.return_value = htsp
+    mock_client.read_cooling_setpoint.return_value = clsp
+    is_active = curr_temp < htsp or curr_temp > clsp
+    mock_client.read_hvac_mode.return_value = ("auto", is_active)
+    mock_client.read_current_temperature.return_value = curr_temp
+    await trigger_polling(hass, freezer)
+    expected_action = (
+        HVACAction.HEATING
+        if curr_temp < 68
+        else HVACAction.COOLING
+        if curr_temp > 72
+        else HVACAction.OFF
+    )
```

Comment:
We should not use `if` in the test code as it makes it unclear what is actually tested. Rather add `expected_action` as a test parameter